src/core/Form/TextInput/TextInput.tsx

```diff
@@ -151,15 +159,25 @@ class BaseTextInput extends Component<TextInputProps> {
         </LabelText>
         <HintText id={this.hintTextId}>{hintText}</HintText>
         <HtmlDiv className={textInputClassNames.inputElementContainer}>
-          <HtmlInput
-            {...passProps}
-            id={this.id}
-            className={textInputClassNames.inputElement}
-            type={type}
-            placeholder={visualPlaceholder}
-            {...{ 'aria-invalid': status === 'error' }}
-            {...getDescribedBy()}
-          />
+          <Debounce waitFor={this.props.debounce || 0}>
+            {(debouncer: Function) => (
+              <HtmlInput
+                {...passProps}
+                id={this.id}
+                className={textInputClassNames.inputElement}
+                type={type}
+                placeholder={visualPlaceholder}
+                {...{ 'aria-invalid': status === 'error' }}
+                {...getDescribedBy()}
+                onChange={(event: ChangeEvent<HTMLInputElement>) => {
+                  const eventValue = event.currentTarget.value;
```

Comment:
this could actually be inside the if statement so when there is no propOnChange the const never gets created.

Reply:
Good point. Moved it inside.